### DISCUSSION TITLE
How do I make vcpkg install the dependencies / libraries alongside my build artifacts in manifest mode when using CMake for building?

### DISCUSSION THREAD

**Discussion by McKillroy** [microsoft/vcpkg#20526] [Q&A] [answered]
We're building a project in manifest mode using a vcpkg.json and a custom registry.
We use CMake for building, since we develop cross platform win/*ix.
So far it works.
But when I install my targets with CMakes install(...) they end up in a different tree then the "vcpkg_installed" dependencies.
How can I fix this in a not-too-hacky way to have everything together in one tree?
(Non hackish in like not just using  `SET(CMAKE_INSTALL_PREFIX ${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET})` )

Thanks
~Chris

**Comment by autoantwort** [comment]
Why do you want to install to `vcpkg_installed`?

**Reply by McKillroy** [reply]
Because I want to have everything self contained in one single tree.

**Reply by autoantwort** [reply]
I think this is a bad idea, you should never write own stuff to the `vcpkg_installed` folder.

**Reply by dg0yt** [reply]
> I want to have everything self contained in one single tree.

This would also include tools to build your software.

**Comment by dg0yt** [comment, marked as answer]
For Windows, vcpkg goes the other way and copies required DLLs into your build directory.
(I know it is getting more complicated if you also want to use data, e.g. for `proj` or `gdal`.)

**Reply by dg0yt** [reply]
Additions:
- The "app-local" deployment to the build directory is also done for osx triplets. It is enabled by default.
- Experimental "app-local" deployment to the install directory (i.e. cmake project setting, not vcpkg's `installed`) can be enabled for windows and uwp via X_VCPKG_APPLOCAL_DEPS_INSTALL.

**Reply by McKillroy** [reply]
That finally worked very nicely and saved my day :)

`set(X_VCPKG_APPLOCAL_DEPS_INSTALL ON)`

**Comment by McKillroy** [comment]
I feel the suggested answers to not yet really solve the issue.

I guess a vcpkg function like a fictional `VCPKG_INSTALL_DEPENDENCIES(target)` would be nice to have.

Seems I'm back to hacking some CMake together here, either
`install(IMPORTED_RUNTIME_ARTIFACTS  ...`  (for some reason it seems not to work on all my dependencies / transient deps) or
`install(DIRECTORY ...` (installs empty directories when applying a pattern)
ore using file globbing.

Are there better ways to do it?
Should I make a feature request for this?
